[WIP] IME preedit

Gillibald opened the pull request:
## What does the pull request do?
<!--- Give a bit of background on the PR here, together with links to with related issues etc. -->


## What is the current behavior?
<!--- If the PR is a fix, describe the current incorrect behavior, otherwise delete this section. -->


## What is the updated/expected behavior with this PR?
<!--- Describe how to test the PR. -->


## How was the solution implemented (if it's not obvious)?
<!--- Include any information that might be of use to a reviewer here. -->


## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Breaking changes
<!--- List any breaking changes here. -->

## Obsoletions / Deprecations
<!--- Obsolete and Deprecated attributes on APIs MUST only be included when discussed with Core team. @grokys, @kekekeks & @danwalmsley -->

## Fixed issues
<!--- If the pull request fixes issue(s) list them like this: 
Fixes #123
Fixes #456
-->
Fixes: #8336

Gillibald committed:
Rendering part of PreeditText

Gillibald committed:
Imm32 preedit

Gillibald committed:
Cleanup

Gillibald committed:
Merge remote-tracking branch 'upstream/master' into feature/ImePreeditText

avaloniaui-team commented:
You can test this PR using the following package version. `0.10.999-cibuild0023297-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

Gillibald committed:
Merge remote-tracking branch 'upstream/master' into feature/ImePreeditText

Gillibald committed:
Android preedit

Gillibald committed:
Adjust remaining InputMethodImpl

Gillibald committed:
Merge branch 'master' into feature/ImePreeditText

Commit:
Merge branch 'master' into feature/ImePreeditText

Gillibald committed:
Fix iOS build

Gillibald committed:
Merge remote-tracking branch 'origin/feature/ImePreeditText' into feature/ImePreeditText

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0023478-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

Gillibald committed:
Disable special preedit highlight

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0023480-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

Commit:
add simple mobile sandbox apps.

Commit:
Merge branch 'feature/ImePreeditText' of github.com:Gillibald/Avalonia into feature/ImePreeditText

Commit:
Merge branch 'master' into feature/ImePreeditText

Commit:
Merge remote-tracking branch 'origin/master' into feature/ImePreeditText

Commit:
whitespace.

Commit:
add implementation of advanced ios ime api.

Commit:
remove some assets.

Commit:
add implementation of returnkeytype for ios.

Commit:
add hack implementation of surrounding text.

Commit:
further ios ime implementation.

Commit:
make implementation consistent with ios sample.

https://developer.apple.com/library/archive/samplecode/SimpleTextInput/Introduction/Intro.html#//apple_ref/doc/uid/DTS40010633
